draftjs_exporter/composite_decorators.py

```diff
@@ -0,0 +1,60 @@
+from __future__ import absolute_import, unicode_literals
+
+import cgi
+import re
+
+
+class CompositeDecorator():
+
+    # Override this to describe the pattern to be replaced
+    SEARCH_RE = re.compile('$^')
+
+    # Implemente this to generate a html string to replace the matched text
+    def replace(self, match):
+        raise NotImplementedError
+
+    def process(self, text, parent=None):
+        self.parent = parent
+        result = ""
+        begin = final = 0
+        for i in self.SEARCH_RE.finditer(text):
+            begin, end = i.span()
+            # append other content
+            result += text[final:begin]
+            # append replaced content
+            result += self.replace(i)
+            final = end
+
+        result += text[final:]
+        return result
+
+
+class URLDecorator(CompositeDecorator):
+    """
+    replace url in plain text to actual html link
+    """
```

Comment:
Not that big of a deal but when we merge this I think we will want to move this to another file. Right now the distinction between "the class that is meant to be implemented" and "a particular implementation" isn't clear enough.

For comments, they should be sentences starting with a capital and ending with a period.